static/app/stores/sidebarProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import {createContext, useContext, useReducer} from 'react';
+
+function assertNever(x: never) {
+  throw new TypeError('Unhandled sidebar reducer action: ' + x);
+}
+
+import {SidebarPanelKey} from 'sentry/components/sidebar/types';
+
+type State = SidebarPanelKey | '';
+
+type HidePanelAction = {type: 'hide panel'};
+type TogglePanelAction = {payload: SidebarPanelKey; type: 'toggle panel'};
+type ActivatePanelAction = {payload: SidebarPanelKey; type: 'activate panel'};
+type SidebarAction = HidePanelAction | TogglePanelAction | ActivatePanelAction;
+
+const SidebarContext = createContext<[State, React.Dispatch<SidebarAction>] | null>(null);
+
+export function SidebarReducer(state: State, action: SidebarAction) {
+  switch (action.type) {
+    case 'activate panel':
+      return action.payload;
+    case 'hide panel':
+      return '';
+    case 'toggle panel':
+      if (state === action.payload) {
+        return '';
+      }
+      return action.payload;
+    default:
+      assertNever(action);
+      return state;
+  }
+}
+
+interface Props {
+  children: React.ReactNode;
+  initialPanel?: State;
+}
+
+export function SidebarProvider(props: Props) {
+  const reducer = useReducer(SidebarReducer, props.initialPanel ?? '');
```

Comment:
Do we really need a reducer here?

This store is only for showing / hiding the sidebar. Why can't we just use a 

```tsx
const [showSidebar, setShowSidebar] = useState(...)
```

Reply:
👍 to generally keeping to state where possible since it in basic cases it's simpler code, if we can remove a couple actions when porting stores and simplify to just state that's def better... however, in general replacing these stores generically with context + reducer makes sense from an ease-of-migration standpoint, so imo this is fine.

Reply:
Yeah +1 here with @k-fish. I didnt want to refactor the actual functionality - I think we should do that later.